FirebaseRemoteConfig/SwiftNew/ConfigRealtime.swift

```diff
@@ -0,0 +1,665 @@
+                         withStatus: .failure,
+                         withError: NSError(domain: RemoteConfigErrorDomain,
+                                            code: RemoteConfigError.internalError.rawValue,
+                                            userInfo: [
+                                              NSLocalizedDescriptionKey: errorDescription,
+                                            ]))
+        return
+      }
+      /// We have a valid token. Get the backing installationID.
+      installations.installationID { [weak self] identifier, error in
+        guard let self = self else { return }
+        // Dispatch to the RC serial queue to update settings on the queue.
+        self.realtimeLockQueue.async {
+          self.settings.configInstallationsToken = tokenResult.authToken
+          self.settings.configInstallationsIdentifier = identifier ?? ""
+          if let error = error {
+            let errorDescription = "Error getting iid : \(error)."
+            RCLog.error("I-RCN000055", errorDescription)
+            self.isRequestInProgress = false
+            var userInfo = [String: Any]()
+            userInfo[NSLocalizedDescriptionKey] = errorDescription
+            userInfo[NSUnderlyingErrorKey] = error
+            self.reportCompletion(
+              onHandler: completionHandler,
+              withStatus: .failure,
+              withError: NSError(domain: RemoteConfigErrorDomain,
+                                 code: RemoteConfigError.internalError.rawValue,
+                                 userInfo: userInfo)
+            )
+          } else if let identifier = identifier {
+            RCLog.info("I-RCN000022", "Success to get iid : \(identifier)")
+            self.reportCompletion(onHandler: completionHandler,
+                                  withStatus: .noFetchYet,
+                                  withError: nil)
+          }
+        }
+      }
+    }
+  }
+
+  @objc public
+  func createRequestBody(completion: @escaping (Data) -> Void) {
+    refreshInstallationsToken { status, error in
+      if self.settings.configInstallationsIdentifier.isEmpty {
+        RCLog.debug(
+          "I-RCN000013",
+          "Installation token retrieval failed. Realtime connection will not include " +
+            "valid installations token."
+        )
+      }
+      var request = self.request
+      request.setValue(self.settings.configInstallationsToken,
+                       forHTTPHeaderField: installationsAuthTokenHeaderName)
+      if let etag = self.settings.lastETag {
+        request.setValue(etag, forHTTPHeaderField: ifNoneMatchETagHeaderName)
+      }
+
+      let postBody = """
+      {
+      project:'\(self.options.gcmSenderID)',
+      namespace:'\(Utils.namespaceOnly(self.namespace))',
+      lastKnownVersionNumber:'\(self.configFetch.templateVersionNumber)',
+      appId:'\(self.options.googleAppID)',
+      sdkVersion:'\(RemoteConfig.sdkVersion())',
```

Comment:
In Objc, this was `Device.remoteConfigPodVersion`

Reply:
Fixed in my next commit.